### ISSUE TITLE
Specifying "server_attributes"

### ISSUE THREAD

**Issue by Stephan202** [scalyr/scalyr-logback#9]
When using the Scalyr Agent, in `/etc/scalyr-agent-2/agent.json` one can define zero or more `server_attributes`. After a cursory inspection of the `scalyr-logback` implementation it seems to me this project does not provide equivalent functionality. Did I perhaps miss this, or would it be possible to add such support? For example, the [`logstash-logback-encoder`](https://github.com/logstash/logstash-logback-encoder) allows one to configure "Global Custom Fields" inside `logback.xml`; that's exactly the kind of thing I'm looking for.

**Comment by steve-scalyr** [comment]
When initializing the Scalyr library, try adding an EventAttributes
argument to the Events.init call:

Events.init("...API key...", maxBufferRam, null,
            new EventAttributes("field1", "value1", "field2", "value2", ...)
);


This argument functions in the same way as the agent's server_attributes.

Cheers,
Steve

**Comment by Stephan202** [comment]
@steve-scalyr, thanks, that works. Though it does mean duplicating the API key etc. inside the code, at a point before Logback is initialized. (The settings specified in `logback.xml` will basically be ignored, because second and subsequent calls to `Events#init` are NO-OPs.)

I'm not so familiar with the Logback internals, but I might try to contribute config-level support for this feature. Would you guys be interested? Also, for us to switch to this appender we'd need some of the functionality proposed in #6; are there plans that get that PR or something equivalent merged?

**Comment by steve-scalyr** [comment]
Yes, it would be great to have configuration support for server attributes
– if you'd like to contribute that, we'd be happy to merge it. Thanks!

#6 <https://github.com/scalyr/scalyr-logback/pull/6> has some nice
features, but since it changes existing behavior, it probably needs to be
tweaked to make the new features optional. I'm not sure why the PR has been
languishing.

Cheers,
Steve
